src/sgame/sg_bot_util.cpp

```diff
@@ -1355,6 +1358,31 @@ float BotAimAngle( gentity_t *self, vec3_t pos )
 	return AngleBetweenVectors( forward, ideal );
 }
 
+// fills allyEntityNumbers with players of a team, and return the number of entities found.
+// pre-condition:
+// * team is playable
+// * allyEntityNumbers is an array of at least maxAllies unsigned integers
+// * allyEntityNumbers memory is initialized
+// * there are less than MAX_CLIENTS players in team
+// * there is enough space to store all team's player's references into allyEntityNumbers
+unsigned int FindAllies( unsigned int *allyEntityNumbers, unsigned int maxAllies, team_t team )
```

Comment:
I think it would help to use `unsigned int allyEntityNumbers[]` or even `unsigned int allyEntityNumbers[maxAllies]` here, though the later could exist in C only (not C++).

Reply:
Also, shouldn't this should take itself as argument to avoid including itself? Then `self->ent.OnSameTeam(other)` or `G_OnSameTeam` could be used.